### PR TITLE
[Do not merge] Proposed Docs Reorg

### PR DESCRIPTION
**Pull request by plumbis**
This PR is a draft for the docs reorg discussed in [crossplane/crossplane #3398](https://github.com/crossplane/crossplane/issues/3398).

https://deploy-preview-254--crossplane.netlify.app/

This proposes a few major changes, including:
* The creation of a "[Knowledge Base](https://deploy-preview-254--crossplane.netlify.app/knowledge-base/)" for content that is not version specific and can be more in depth on a single topic or cross multiple topics.
* The creation of a "[Developer Guide](https://deploy-preview-254--crossplane.netlify.app/contribute/)" to document existing contribution process and rules for both Crossplane source and Crossplane docs.
* A docs "[landing page](https://deploy-preview-254--crossplane.netlify.app/)" to act as a point of entry for accessing the User Docs, KB or Developer Guide
* A [significant reorganization](https://deploy-preview-254--crossplane.netlify.app/v1.11/) of files within a docs version. Use `v1.11` as the version to see the proposed organization.
    * The diff of files between the current docs and the proposal is here: https://deploy-preview-254--crossplane.netlify.app/v1.11/file-map/

The proposed organization is focused on the existing content and does not map directly to the outline in  [crossplane/crossplane #3398](https://github.com/crossplane/crossplane/issues/3398). 

At this point no work has been done on some of the proposed content merge. Do not focus on the current content of the pages.

### CODE REVIEW

**Comment by negz** [comment]
> The creation of a "[Knowledge Base](https://deploy-preview-254--crossplane.netlify.app/knowledge-base/)" for content that is not version specific and can be more in depth on a single topic or cross multiple topics.

How will this relate to https://github.com/crossplane/crossplane/discussions (which includes a knowledge base)? Do you expect this to replace it?

**Comment by plumbis** [comment]
Yes. Most of those entries are either notes (e.g., https://github.com/crossplane/crossplane/discussions/2206) or are questions/discussions (e.g., [#1](https://github.com/crossplane/crossplane/discussions/2475), [#2](https://github.com/crossplane/crossplane/discussions/3122), [#3](https://github.com/crossplane/crossplane/discussions/2779))

The new knowledge base would be curated articles of high quality. The docs also provide a better reader experience because of the [styling features](https://deploy-preview-254--crossplane.netlify.app/contribute/docs/features/) and better discoverability since they are baked into the user docs and part of the docs search.

File: content/contribute/coding/add-secret-store.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
+---
+title: Adding Secret Stores
```
Comment (negz) [inline content/contribute/coding/add-secret-store.md:2]: Not sure how to handle this, but I'd like to keep stuff relating to alpha or beta features very clearly delineated from everything else - under their own section, or some kind of very obvious banner/header?

File: content/contribute/_index.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
+---
```
Comment (negz) [inline content/contribute/_index.md:1]: All this "how to work on Crossplane" stuff does feel like it probably belong in the c/c repo more than the docs repo. Maybe we move it back to a `contributing` or `contributing-docs` dir there and link from here?

File: content/knowledge-base/guides/composition-revisions.md [hunk 3]
```diff
@@ -0,0 +1,153 @@
+---
+title: Composition Revisions
+---
```
Comment (negz) [inline content/knowledge-base/guides/composition-revisions.md:3]: Dropping a note that this is a beta feature and should be tagged clearly as such.

File: content/knowledge-base/guides/learn_more.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
+---
+title: Learn More
+weight: 307
+---
```
Comment (negz) [inline content/knowledge-base/guides/learn_more.md:4]: Nit - I think we mostly use hyphens in filenames. Would be nice to not use an underscore in this one (can change in future).

**Review by negz — APPROVED** [review]
Few nits (most of which probably applied to the existing content so feel free to ignore). Otherwise the proposed reorg makes sense to me.

File: content/knowledge-base/guides/composition-revisions.md (hunk 3 above)
Comment (plumbis) [reply content/knowledge-base/guides/composition-revisions.md:3]: added in a unique PR. https://github.com/crossplane/docs/pull/263

File: content/contribute/coding/add-secret-store.md (hunk 1 above)
Comment (plumbis) [reply content/contribute/coding/add-secret-store.md:2]: added in a unique PR. https://github.com/crossplane/docs/pull/263

File: content/contribute/_index.md (hunk 2 above)
Comment (plumbis) [reply content/contribute/_index.md:1]: All of this has been removed and the contrib guide is only about contributing to the docs.

File: content/knowledge-base/guides/learn_more.md (hunk 4 above)
Comment (plumbis) [reply content/knowledge-base/guides/learn_more.md]: This isn't from a change, there are multiple legacy files that use underscores. This particular page comes from `/content/<version>/reference/learn_more.md`

**Pull request closed without merging**